presto-main/src/main/java/com/facebook/presto/execution/buffer/PartitionedOutputBuffer.java

```diff
@@ -199,6 +215,30 @@ public void enqueue(int partitionNumber, List<SerializedPage> pages)
         serializedPageReferences.forEach(SerializedPageReference::dereferencePage);
     }
 
+    @Override
+    public void setNoMorePagesForLifespan(Lifespan lifespan)
+    {
+        noMorePagesForLifespan.add(lifespan);
+    }
+
+    @Override
+    public void registerLifespanCompletionCallback(Consumer<Lifespan> callback)
+    {
+        checkState(lifespanCompletionCallback == null);
```

Comment:
Add the messages as in the `ArbitraryOutputBuffer#registerLifespanCompletionCallback`